src/learning/utils.py

```diff
@@ -0,0 +1,51 @@
+import torch
+import torchvision.transforms as transforms
+
+
+class CycleData():
+    def __init__(self, discr_x, discr_y, gen_xy, gen_yx, batch_x, batch_y):
+        self.batch_x = batch_x
+        self.batch_y = batch_y
+        self.synthesis_x = gen_yx(batch_y)
+        self.synthesis_y = gen_xy(batch_x)
+        self.batch_x_predictions = discr_x(self.batch_x)
+        self.batch_y_predictions = discr_y(self.batch_y)
+        self.synthesis_x_predictions = discr_x(self.synthesis_x)
+        self.synthesis_y_predictions = discr_y(self.synthesis_y)
+        self.id_x_approximations = gen_yx(self.synthesis_y)
+        self.id_y_approximations = gen_xy(self.synthesis_x)
+
+
+def get_transform(image_size):
+    return transforms.Compose([
+        transforms.Resize(image_size),
+        transforms.CenterCrop(image_size),
+        transforms.ToTensor(),
+        transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
+
+
+def get_batch_data(data_x_loader, data_y_loader, discr_x, discr_y, gen_xy,
+                   gen_yx, device):
+    batch_x, _ = next(iter(data_x_loader))
+    batch_x = batch_x.to(device)
+    batch_y, _ = next(iter(data_y_loader))
+    batch_y = batch_y.to(device)
+    cycle_data = CycleData(discr_x, discr_y, gen_xy, gen_yx, batch_x, batch_y)
+    return cycle_data
+
+
+def init_weights_gaussian(model):
+    classname = model.__class__.__name__
+    if classname.find('Conv') != -1:
```

Comment:
What happens to ConvLayers?

Reply:
Their weights are initialized according to a Gaussian distribution.
Other layers, such as ReLU or Normalization layers do not have weights that are meant to be initialized.

Reply:
Ah, I see. 
I think normalization layers actually do have weights to initialise, i.e. the scaling factor and the offset.

Reply:
Very good point.
My understanding is that InstanceNorm2d is instantiated without learnable weights by default.